app/scripts/controller-init/confirmations/transaction-controller-init.ts

```diff
@@ -153,6 +157,15 @@ export const TransactionControllerInit: ControllerInitFunction<
           transactionMeta,
           rawTx,
         ),
+      publishBatch: async (_request: PublishBatchHookRequest) =>
+        await publishBatchSmartTransactionHook({
+          transactionController: controller,
+          smartTransactionsController: smartTransactionsController(),
```

Comment:
Same here, not needed now but ideally we'd avoid passing controller instances around, and rely on the messenger to communicate with them.